src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java

```diff
@@ -1,587 +1,834 @@
+                .returnType(BitbucketProjectData.class))
+        .thenReturn(bitbucketProjectData);
+    // bitbucket aus repo creation - oc-config
+    RepositoryData bitbucketRepositoryDataOCConfig =
+        readTestData("bitbucket-create-repo-occonfig-response", RepositoryData.class);
+    Repository occonfigRepo = new Repository();
+    occonfigRepo.setName(bitbucketRepositoryDataOCConfig.getName());
+    occonfigRepo.setUserGroup(userGroup);
+    occonfigRepo.setAdminGroup(adminGroup);
+
+    // bitbucket aux repo creation - design repo
+    RepositoryData bitbucketRepositoryDataDesign =
+        readTestData("bitbucket-create-repo-design-response", RepositoryData.class);
+    Repository designRepo = new Repository();
+    designRepo.setName(bitbucketRepositoryDataDesign.getName());
+    designRepo.setUserGroup(userGroup);
+    designRepo.setAdminGroup(adminGroup);
+    //    Mockito.when(
+    //            mockOldRestClient.callHttp(
+    //                contains(realBitbucketAdapter.getAdapterApiUri()),
+    //                refEq(occonfigRepo, "adminGroup", "userGroup"),
+    //                anyBoolean(),
+    //                eq(HttpVerb.POST),
+    //                eq(RepositoryData.class)))
+    //        .thenReturn(bitbucketRepositoryDataOCConfig);
+    mockHelper
+        .mockExecute(
+            matchesClientCall()
+                .url(containsString(realBitbucketAdapter.getAdapterApiUri() + "/TESTP/repos"))
+                // .bodyMatches(samePropertyValuesAs(occonfigRepo))
+                .method(HttpMethod.POST)
+                .returnType(RepositoryData.class))
+        .thenReturn(bitbucketRepositoryDataOCConfig, bitbucketRepositoryDataDesign);
+
+
+
+    // will cause cleanup
+    String rundeckUrl = realRundeckAdapter.getAdapterApiUri() + "/project/";
+    if (fail) {
+      //      Mockito.when(
+      //              mockOldRestClient.callHttpTypeRef(
+      //                  contains(rundeckUrl), anyMap(), anyBoolean(), eq(HttpVerb.GET), any()))
+      //          .thenThrow(new IOException("Rundeck TestFail"));
+      mockHelper
+          .mockExecute(
+              matchesClientCall()
+                  .url(containsString(rundeckUrl))
+                  .bodyMatches(instanceOf(Map.class))
+                  .method(HttpMethod.GET))
+          .thenThrow(new IOException("Rundeck TestFail"));
+    } else {
+      //      Mockito.when(
+      //              mockOldRestClient.callHttpTypeRef(
+      //                  contains(rundeckUrl), anyMap(), anyBoolean(), eq(HttpVerb.GET), any()))
+      //          .thenReturn(jobList);
+      // populate the rundeck jobs
+      mockRundeckDefaultJobs();
+    }
+
+    // rundeck create-projects job execution
+    ExecutionsData execution =
+        readTestData("rundeck-create-project-response", ExecutionsData.class);
+
+    String createJobId = realJobStore.getJobIdForJobName("create-projects");
+    assertNotNull(createJobId);
+
+    //    Mockito.when(
```

Comment:
Is the call still necessary?

Reply:
Ok, please see my last commits on this PR.